### PR TITLE
Rewrite the knowl javascript with code that uses a bootstrap collapsible.

### PR DESCRIPTION
**Pull request by drgrice1**
The real code for that is in openwebwork/pg#638.  This just adds the defered attribute to the loading of the javascript as is needed for the new code.

It also moves the loading of the knowl.js file from the output_JS method in GatewayQuiz.pm to the gateway.template file.  It didn't make sense that the knowl.css style was loaded in the template, but the javascript in the output_JS method.  That method of the GatewayQuiz.pm module should really only be used for things that are conditionally loaded depending on settings and such, and not things that are always loaded.  At least for GatewayQuiz.pm which has a dedicated template file.

### CODE REVIEW

File: lib/WebworkClient/simple_format.pl [hunk 1]
```diff
@@ -27,7 +27,8 @@
 <script src="/webwork2_files/js/apps/InputColor/color.js"></script>
 <script src="/webwork2_files/js/apps/Base64/Base64.js"></script>
 <script src="/webwork2_files/js/vendor/underscore/underscore.js"></script>
-<script src="/webwork2_files/js/apps/Knowls/knowl.js"></script>
+<script src="/webwork2_files/js/apps/Knowls/knowl.js" defer></script>
+>>>>>>> Rewrite the knowl javascript with code that uses a bootstrap collapsible.
```
Comment (pstaabp) [inline lib/WebworkClient/simple_format.pl:31]: Is this supposed to be a comment? or a leftover merge conflict?

Comment (drgrice1) [reply lib/WebworkClient/simple_format.pl]: Yeah, that was a mistake.  It is fixed now.

**Comment by pstaabp** [comment]
I'm getting a lot of file-not found errors in the js console.  For example:
```
localhost/:54 GET http://localhost/webwork2_files/js/apps/MathQuill/mathquill.css net::ERR_ABORTED 404 (Not Found)
localhost/:54 GET http://localhost/webwork2_files/js/apps/MathQuill/mqeditor.css net::ERR_ABORTED 404 (Not Found)
localhost/:54 GET http://localhost/webwork2_files/js/apps/ImageView/imageview.css 
```

I did `npm install` in both webwork2 and pg and restarted the server.  This branch doesn't seem to have these files in them.  Maybe this needs a merge from another branch?

**Comment by drgrice1** [comment]
This should not have those files.  Did you check out openwebwork/pg#638?

**Comment by pstaabp** [comment]
> This should not have those files. Did you check out [openwebwork/pg#638](https://github.com/openwebwork/pg/pull/638)?

Yes.  I had some leftover files that I didn't stash from a previous PR, but still have errors.  Looks like on line 2461 and 2462 of `GatewayQuiz.pm` have the following lines:
```
		print qq{<link href="$site_url/js/apps/MathQuill/mathquill.css" rel="stylesheet" />};
		print qq{<link href="$site_url/js/apps/MathQuill/mqeditor.css" rel="stylesheet" />};
```
and there definitely isn't any MathQuill directory in `htdocs/js/apps`.

**Comment by drgrice1** [comment]
That is the same thing.  Those files are not in `webwork2/htdocs/js/apps`.  They are in `pg/htdocs/js/apps`.  The links in `GatewayQuiz.pm` are correct.

**Comment by drgrice1** [comment]
Also, have you updated your conf/webwork.apache2.4-config file with the changes from PR #1536?

**Comment by pstaabp** [comment]
> Also, have you updated your conf/webwork.apache2.4-config file with the changes from PR #1536?

This was the issue.  I thought I had from previous PRs.

**Comment by pstaabp** [comment]
This looks great in HomeworkSets (as a student) and ProblemSetDetail, LibraryBrowser looking in solutions.  

Also, knowls is in some of the help links, correct?  Most that I am finding open a new tab.  We talked about this.  Did you change all functionality like this?

**Comment by drgrice1** [comment]
Here is a test problem that demonstrates many of the knowl uses.  It contains a helpLink.

[knowls.pg.txt](https://github.com/openwebwork/webwork2/files/7906341/knowls.pg.txt)

**Comment by drgrice1** [comment]
Note that the `helpLink` method is defined in `PGbasicmacros.pl`, and it just calls the `knowlLink` method.  So there was no need to modify the `helpLink` method directly.

**Comment by drgrice1** [comment]
In fact, all uses of knowls are done through the `knowlLink` method.  So that is why the only modification is to the `knowlLink` method other than removing the base64 encoding of hints and solutions, and a little clean up in the `sub HINT` routine.

**Review by pstaabp — APPROVED** [review]
Help Links look good too.

**Pull request merged**